### PR TITLE
Remove support for deprecated rpc endpoints

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
solana-rpc is carrying around a lot of obsolete and deprecated endpoints. v2.0 means it is time to remove them.

#### Summary of Changes

- Remove obsolete and deprecated rpc modules
- Remove validator `--enable-rpc-obsolete_v1_7` flag
- Update changelog

One piece of follow-up work can be to remove `solana_rpc_client_api::deprecated_config`

Also, I decided to handle `rpc_deprecated_v1_18` in a separate PR, because it looks like the client methods weren't deprecated at the same time 😞 Will think on how to handle.

### CODE REVIEW

**Comment by CriesofCarrots** [comment]
@willhickey I added you in case you have opinions on whether I've horribly abused the changelog 😅

**Comment by CriesofCarrots** [comment]
Rebased in order to put changelog entries under new v2.1 section

**Comment by joncinque** [comment]
cc @steveluscher to check if anything needs to be done in web3.js

**Comment by steveluscher** [comment]
ty! https://github.com/solana-labs/solana-web3.js/issues/2859

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,14 @@ Release channels have their own copy of this changelog:
 
 <a name="edge-channel"></a>
 ## [2.1.0] - Unreleased
+* Breaking
```
Comment (t-nelson) [inline CHANGELOG.md:21]: are we breaking semver here? 2.0.0 and 2.1.0 are to be compatible to adhere afaik

Comment (CriesofCarrots) [reply CHANGELOG.md:21]: I can certainly backport to v2.0 as well, if you think that's better.
If removals like this *have* to happen in the vX.0.0 release, we definitely need to provide more warning to engineers before branching than we did with v2.0.0.

Comment (CriesofCarrots) [reply CHANGELOG.md:21]: Also, fwiw, I had this PR up before the branch happened, but didn't receive reviews.

Comment (willhickey) [reply CHANGELOG.md:21]: I vote we proceed and backport to v2.0

Comment (t-nelson) [reply CHANGELOG.md:21]: hmm... we should probably make deprecated api removal a blocker for major version branching in the future

i'm not opposed to a 2.0.0 bp. or at least i'm less opposed to that than removing deprecated api in a minor. @willhickey wdyt?

Comment (CriesofCarrots) [reply CHANGELOG.md:21]: > I vote we proceed and backport to v2.0

Okay, works for me. I will update the changelog in this PR to reflect that.
I will have ~3 follow-up PRs, which will need similar treatment.

**Review by t-nelson — APPROVED** [review]

**Pull request merged**